### PR TITLE
feat: adding smock package

### PR DESCRIPTION
**Pull request by wei3erHase**

### CODE REVIEW

File: test/unit/BPool.t.sol [hunk 1]
```diff
@@ -340,8 +334,15 @@ contract BPool_Unit_JoinPool is BasePoolTest {
...
     // Set balances
     for (uint256 i = 0; i < tokens.length; i++) {
-      _setRecordBound(tokens[i]);
-      _setRecordBalance(tokens[i], _fuzz.balance[i]);
+      _setRecord(
+        tokens[i],
+        BPool.Record({
+          bound: true,
+          index: i,
+          denorm: BONE, // TODO: set denorm? is it relevant?
```
Comment (0xAustrian) [inline test/unit/BPool.t.sol:342]: Not relevant, at least in JoinPool

File: package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
     "lint:sol-logic": "solhint -c .solhint.json 'src/**/*.sol' 'script/**/*.sol'",
     "lint:sol-tests": "solhint -c .solhint.tests.json 'test/**/*.sol'",
     "prepare": "husky install",
+    "smock": "smock-foundry --contracts src/contracts",
```
Comment (0xAustrian) [inline package.json:27]: we should add this when building and running the tests

File: test/unit/BPool.t.sol (hunk 1 above)
Comment (wei3erHase) [reply test/unit/BPool.t.sol:342]: interesting! 👀

File: package.json (hunk 2 above)
Comment (wei3erHase) [reply package.json:27]: mhhh, i've added it on the CI, but i believe smock contracts are sth that are not gonna suffer much changes, and if it does all you get is a compilation error on a smock contract, and you'll know you must run `yarn smock`, but adding this routine to the `yarn build` seems like suboptimal time-wise, wdyt?

File: test/unit/BPool.t.sol [hunk 3]
```diff
@@ -340,8 +334,15 @@ contract BPool_Unit_JoinPool is BasePoolTest {
 
     // Set balances
     for (uint256 i = 0; i < tokens.length; i++) {
-      _setRecordBound(tokens[i]);
-      _setRecordBalance(tokens[i], _fuzz.balance[i]);
+      _setRecord(
+        tokens[i],
+        BPool.Record({
+          bound: true,
+          index: i,
```
[the remark this answers is not in the document]
Comment (wei3erHase) [reply test/unit/BPool.t.sol]: now i realized that i can set both `index` and `denorm` to zero

File: package.json (hunk 2 above)
Comment (0xAustrian) [reply package.json:27]: no worries, I was suggesting it so you can run `yarn test:unit` without having to run `yarn smock` before

**Review by 0xAustrian — APPROVED** [review]

**Review by 0xAustrian — APPROVED** [review]

**Pull request merged**